### PR TITLE
Clean up deprecated APIs in core packages for 3.0

### PR DESCRIPTION
**Pull request by pmconne**

### CODE REVIEW

**Comment by pmconne** [comment]
@RBBentley TerrainSettings locatable flag is marked deprecated, telling callers to use the BackgroundMapSettings locatable flag instead:
```
core\common\src\TerrainSettings.ts|33| /** @deprecated use [[BackgroundMapProps.nonLocatable]] */
core\common\src\TerrainSettings.ts|64| /** @deprecated use [[BackgroundMapSettings.locatable]] */
```
But the BackgroundMapSettings flag is implemented in terms of TerrainSettings' flag. Is this actually deprecated?
```
  public get locatable(): boolean {
    // Handle deprecated TerrainProps.nonLocatable:
    // - If TerrainProps.nonLocatable=true and terrain is on, terrain is not locatable.
    // - Otherwise, use BackgroundMapProps.nonLocatable.
    if (this.applyTerrain && !this.terrainSettings.locatable) // eslint-disable-line deprecation/deprecation
      return false;
```
Do I understand correctly that you don't want people to continue to use the "legacy" terrain props flag, but you will continue to honor it if present?

**Comment by RBBentley** [comment]
Yes - that is correct.  At the time, I was attempting to account for older saved views that may have the terrain non-locatable setting.  I'm not sure if any of those exist from DR - but I expect not.

**Comment by kabentley** [comment]
> @kabentley NativeAppOptions adds only two additional properties, both of which are deprecated, but both of which are used. What do you want to do with them?
> 
> ```
> core\frontend\src\NativeApp.ts|126| * @deprecated Initialize authorization for native applications at the backend
> core\frontend\src\NativeApp.ts|130| * @deprecated Initialize authorization for native applications at the backend
> ```

I think we'll have to leave this here until @calebmshafer and @evelynpreslar-bentley finish the authorization rework.

**Comment by pmconne** [comment]
@Mergifyio update

**Review by mgooding — APPROVED** [review]

**Review by markschlosseratbentley — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

File: core/transformer/src/test/IModelTransformerUtils.ts [hunk 1]
```diff
@@ -387,7 +387,7 @@ export class TransformerExtensiveTestScenario {
     assert.isTrue(displayStyle3d.settings.hasSubCategoryOverride);
     assert.equal(displayStyle3d.settings.subCategoryOverrides.size, 1);
     assert.exists(displayStyle3d.settings.getSubCategoryOverride(subCategoryId), "Expect subCategoryOverrides to have been remapped");
-    assert.isTrue(displayStyle3d.settings.excludedElements.has(physicalObjectId1), "Expect excludedElements to be remapped"); // eslint-disable-line deprecation/deprecation
+    assert.isTrue(new Set<string>(displayStyle3d.settings.excludedElementIds).has(physicalObjectId1), "Expect excludedElements to be remapped"); // eslint-disable-line deprecation/deprecation
```
Comment (MichaelBelousov) [inline core/transformer/src/test/IModelTransformerUtils.ts:390]: shouldn't you be able to remove the eslint-ignore comment now that you're no longer using anything deprecated?
also, do we not have a way to fail on unnecessary eslint ignores?

**Review by MichaelBelousov — CHANGES_REQUESTED** [review]

File: core/transformer/src/test/IModelTransformerUtils.ts (hunk 1 above)
Comment (pmconne) [reply core/transformer/src/test/IModelTransformerUtils.ts:390]: yes, overlooked. no, we don't have a lint rule to complain about unnecessarily disabling lint rules.

Comment (pmconne) [reply core/transformer/src/test/IModelTransformerUtils.ts:390]: fixed.

Comment (MichaelBelousov) [reply core/transformer/src/test/IModelTransformerUtils.ts:390]: this line doesn't appear to have received the fix, no?

Comment (pmconne) [reply core/transformer/src/test/IModelTransformerUtils.ts]: try now.

**Review by MichaelBelousov — APPROVED** [review]

**Pull request merged**